docs/ARCHITECTURE.md

```diff
@@ -407,8 +407,41 @@ solidity
 
         emit ResetRateLimit(_amount);
     }
+
+    /**
+     * @notice Used to enable/disable rate limit.
+     * @param value `true` for enabling rate limit and `false` for disabling.
+     * By default rate limit is disabled.
+    */
+    function updateRateLimitStatus(bool value) external onlyRole(SET_RATE_LIMITER_ROLE) {
+        rateLimitEnabled = value;
+        emit RateLimitStatusUpdated(value);
+    }
 ```
 
+### Reset Rate Limit Mechanism
+
+Even though `resetRateLimitAmount` is permissioned, it is still critical in the context of enabling rate limits in the bridge, so it is important to understand how it works.
+
+The account having the `SET_RATE_LIMITER_ROLE` role can reset the rate limit any time when it is enabled.
+
+If the rate limit is reset before the `rateLimitDuration` ends, then the rate limit amount is updated (for the ERC20 gateway, the rate limit duration is also updated but only takes effect when the previous `rateLimitDuration` ends). However, the end timestamp when the current rate limit duration ends and the current withdrawn amount remain unchanged.
+
+If the rate limit is reset after the `rateLimitDuration` ends, then the rate limit amount is updated along with the end timestamp when the current rate limit duration ends, and the current withdrawn amount is set to 0.
+
+Here are some Do's and Don'ts to keep in mind when calling resetRateLimitAmount.
+
+Do's
+
+- Simulate the `resetRateLimitAmount` function call and a withdrawal after that to ensure the rate limit behavior is as desired.
```

Comment:
we might wanna need a script for simulation.

Reply:
I think we'll be using tenderly for it